monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -4959,6 +4960,21 @@ object Observable extends ObservableDeprecatedBuilders {
   def intervalAtFixedRate(period: FiniteDuration): Observable[Long] =
     new builders.IntervalFixedRateObservable(Duration.Zero, period)
 
+  /** Creates on Observable that emits auto-incremented natural numbers
+    * (longs) at a schedule defined by a `start` and `next` parameters.
+    * Particularly, `next` function is called continuously until it returns
+    * `None`, when the resulting Observable completes. Otherwise, its return
+    * value is used to determine next execution instant, as well as input to
+    * its own subsequent invocation.
+    *
+    * @param start input to the `next` function to produce the first tick instant
+    * @param next function emitting the next tick instants based on the result
+    *             of the previous invocation or the `start` parameter
+    * @tparam A type of the elements fed into the `next` function
+    */
+  def atGivenTimesUnfold[A](start: A)(next: A => Option[(A, Instant)]): Observable[Long] =
```

Comment:
Do you think it makes sense to emit `A` values instead? Could potentially be more general than emitting longs, and longs can be trivially recovered by doing things like `as.zipWithIndex.map(_._2)`.

Reply:
Agree, changed.